selfdrive/controls/controlsd.py

```diff
@@ -444,11 +449,22 @@ def state_control(self, CS):
     a_acc_sol = long_plan.aStart + (dt / LON_MPC_STEP) * (long_plan.aTarget - long_plan.aStart)
     v_acc_sol = long_plan.vStart + dt * (a_acc_sol + long_plan.aStart) / 2.0
 
-    # Gas/Brake PID loop
-    actuators.gas, actuators.brake = self.LoC.update(self.active, CS, v_acc_sol, long_plan.vTargetFuture, a_acc_sol, self.CP)
+    if not self.debug_mode:
+      # Gas/Brake PID loop
+      actuators.gas, actuators.brake = self.LoC.update(self.active, CS, v_acc_sol, long_plan.vTargetFuture, a_acc_sol, self.CP)
+
+      # Steering PID loop and lateral MPC
+      actuators.steer, actuators.steeringAngleDeg, lac_log = self.LaC.update(self.active, CS, self.CP, self.VM, params, lat_plan)
+    else:
+      # in joystick debug mode, create dummy lac_log
+      _, _, lac_log = self.LaC.update(self.active, CS, self.CP, self.VM, params, lat_plan)
```

Comment:
Why not create a special lateral log for debug mode? That would allow you to spot from the logs that debug mode was used.

Reply:
If you're fine with adding a new state to `lateralControlState`, I can do that.